core/storage/skill/gae_models.py

```diff
@@ -0,0 +1,154 @@
+    # The commit message.
+    commit_message = ndb.TextProperty(indexed=False)
+    # The commit_cmds dict for this commit.
+    commit_cmds = ndb.JsonProperty(indexed=False, required=True)
+    # The version number of the skill after this commit.
+    version = ndb.IntegerProperty()
+
+    @classmethod
+    def get_commit(cls, skill_id, version):
+        """Returns the commit corresponding to the given skill id and
+        version number.
+
+        Args:
+            skill_id: str. The id of the skill being edited.
+            version: int. The version number of the skill after the commit.
+
+        Returns:
+            The commit with the given skill id and version number.
+        """
+        return cls.get_by_id('skill-%s-%s' % (skill_id, version))
+
+    @classmethod
+    def get_all_commits(cls, page_size, urlsafe_start_cursor):
+        """Fetches a list of all the commits sorted by their last updated
+        attribute.
+
+        Args:
+            page_size: int. The maximum number of entities to be returned.
+            urlsafe_start_cursor: str or None. If provided, the list of
+                returned entities starts from this datastore cursor.
+                Otherwise, the returned entities start from the beginning
+                of the full list of entities.
+
+        Returns:
+            3-tuple of (results, cursor, more) as described in fetch_page() at:
+            https://developers.google.com/appengine/docs/python/ndb/queryclass,
+            where:
+                results: List of query results.
+                cursor: str or None. A query cursor pointing to the next
+                    batch of results. If there are no more results, this might
+                    be None.
+                more: bool. If True, there are (probably) more results after
+                    this batch. If False, there are no further results after
+                    this batch.
+        """
+        return cls._fetch_page_sorted_by_last_updated(
+            cls.query(), page_size, urlsafe_start_cursor)
+
+
+class SkillSummaryModel(base_models.BaseModel):
+    """Summary model for an Oppia Skill.
+
+    This should be used whenever the content blob of the skill is not
+    needed (e.g. search results, etc).
+
+    A SkillSummaryModel instance stores the following information:
+
+        id, description, language_code, last_updated, created_on, version.
+
+    The key of each instance is the skill id.
+    """
+
+    # The description of the skill.
+    description = ndb.StringProperty(required=True)
+    # The ISO 639-1 code for the language this collection is written in.
+    language_code = ndb.StringProperty(required=True, indexed=True)
+
+    # Time when the skill model was last updated (not to be
+    # confused with last_updated, which is the time when the
+    # skill *summary* model was last updated).
+    skill_model_last_updated = ndb.DateTimeProperty(indexed=True)
```

Comment:
Aren't these and the next two fields required?

Reply:
Yeah, made it required.